ironfish/src/primitives/rawTransaction.ts

```diff
@@ -53,17 +57,54 @@ export class RawTransaction {
     >
   }[] = []
 
-  size(): number {
+  size(senderPublicAddress: string): number {
     let size = 0
+    size += 1 // version
     size += 8 // spends length
     size += 8 // notes length
-    size += 8 // fee
-    size += 4 // expiration
-    size += 64 // signature
+    size += 8 // mints length
+    size += 8 // burns length
+    size += TRANSACTION_FEE_LENGTH // fee
+    size += TRANSACTION_EXPIRATION_LENGTH // expiration
+    size += TRANSACTION_PUBLIC_KEY_RANDOMNESS_LENGTH // public key randomness
+    size += this.spends.length * SPEND_SERIALIZED_SIZE_IN_BYTE
     size += this.outputs.length * NOTE_ENCRYPTED_SERIALIZED_SIZE_IN_BYTE
-    size += this.mints.length * (ASSET_LENGTH + 8)
+    size +=
+      this.mints.length *
+      (PROOF_LENGTH + ASSET_LENGTH + AMOUNT_VALUE_LENGTH + TRANSACTION_SIGNATURE_LENGTH)
     size += this.burns.length * (ASSET_ID_LENGTH + 8)
-    size += this.spends.length * SPEND_SERIALIZED_SIZE_IN_BYTE
+    size += TRANSACTION_SIGNATURE_LENGTH // signature
+
+    // Each asset might have a change note, which would need to be accounted for
+    const assetTotals = new BufferMap<bigint>()
+    for (const mint of this.mints) {
+      const asset = new Asset(senderPublicAddress, mint.name, mint.metadata)
```

Comment:
Will creating an asset here have any performance implications?

It looks like we only need the ID here as a way to group mints of the same asset, so I'm wondering if there's another way around this.

Reply:
It will run a few blake2s hashes and check for validity, but it's a fairly cheap check on its own. It's not _ideal_, but I think it's worth the hit to get an exact size. I think the alternative is a best-guess where we might be wrong since we can't calculate if change notes will be created or not.

Ran a quick test on my laptop:
```
Elapsed: 291.958µs
Elapsed: 591.791µs
Elapsed: 235.958µs
Elapsed: 118.75µs
Elapsed: 236.208µs
```

Code:
``` rust
for i in 0..5 {
    let start = Instant::now();
    let a = Asset::new(public_address, &format!("hello {}", i), "metadata").unwrap();
    let elapsed = start.elapsed();
    println!("Elapsed: {:?}", elapsed);
}
```